extra/training.py

```diff
@@ -42,17 +42,19 @@ def train(model, X_train, Y_train, optim, steps, BS=128, device=Device.CPU, loss
     accuracies.append(accuracy)
     t.set_description("loss %.2f accuracy %.2f" % (loss, accuracy))
 
-def evaluate(model, X_test, Y_test, num_classes=None, device=Device.CPU, BS=128):
+def evaluate(model, X_test, Y_test, num_classes=None, device=Device.CPU, BS=128, return_predict=False):
   Tensor.training = False
-  def numpy_eval(num_classes):
+  def numpy_eval(num_classes, return_predict):
     Y_test_preds_out = np.zeros(list(Y_test.shape)+[num_classes])
-    for i in trange(len(Y_test)//BS, disable=os.getenv('CI') is not None):
+    for i in trange((len(Y_test)-1)//BS+1, disable=os.getenv('CI') is not None):
       Y_test_preds_out[i*BS:(i+1)*BS] = model.forward(Tensor(X_test[i*BS:(i+1)*BS], device=device)).cpu().data
     Y_test_preds = np.argmax(Y_test_preds_out, axis=-1)
+    if return_predict:
```

Comment:
Let's change this to not pass in Y at all and just return the predictions always.

Reply:
Are you sure because it is used everywhere in tests and examples? We could also pass it in as an optional variable or just get rid of this feature here in transformers, it was really more for debugging.

Reply:
Yea, we should change it.

Reply:
I've been misunderstanding. Changed it! Like this?